modal/_utils/function_utils.py

```diff
@@ -466,7 +466,10 @@ async def _process_result(result: api_pb2.GenericResult, data_format: int, stub,
 
     if result.status == api_pb2.GenericResult.GENERIC_STATUS_TIMEOUT:
         raise FunctionTimeoutError(result.exception)
-    elif result.status == api_pb2.GenericResult.GENERIC_STATUS_INTERNAL_FAILURE:
+    elif result.status in [
+        api_pb2.GenericResult.GENERIC_STATUS_INTERNAL_FAILURE,
+        api_pb2.GenericResult.GENERIC_STATUS_TERMINATED,
```

Comment:
We also use terminated status code when the app stops. In that case, we don't want the client to retry. Should worker preemptions return INTERNAL_FAILURE instead or something?

Reply:
Alternatively we can stop returning TERMINATED when the function is stopped?

Reply:
Yikes, my bad for having this retry on TERMINATED! As just discussed IRL, that was not my intent - will fix, thanks for catching!